### ISSUE TITLE
Fix categorical losses

### ISSUE THREAD

**Issue by Reid-Taylor** [relflow/relflow#29]
Categorical variables are embedded with an online vocabulary, and because of the constant process of learning new vocabulary entries, the cross entropy loss consistently increases during the initial epochs of pre-training. Heavily imbalanced data among categorical variables prevents us from applying a heuristic hot-fix such as exclusion of losses considered when computing validation loss for a early stopping callback, and few options are immediately present, intuitive, and clean to implement. 

A warm start for a learned vocabulary is helpful, but that is not presently available for all cases, and so we must identify a generalizable solution which does not degrade performance, and which does not rely on any assumptions. 

I hypothesize that a algorithmic approach in which the tokenizer embedding vectors are all of unit magnitude, and in which the loss is of cosine similarity, will help to standardize the loss experienced, while also indirectly encouraging the model to learn more semantically "similar" embeddings for categorical values which are less discernible (and therefore more closely related).
